### PR TITLE
chore: add API docs for role invites/accept role invites

### PR DESCRIPTION
**Pull request by MeghalBisht**
Fixes part of https://github.com/fossasia/open-event-server/issues/8052
Fixes https://github.com/fossasia/open-event-server/issues/5775

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: docs/api/blueprint/role/role_invites.apib [hunk 1]
```diff
@@ -233,3 +233,59 @@ Delete a single role invite.
...
+
+## Accept Role Invite [/v1/role_invites/accept-invite]
+
+### Accept Role Invite using Token [POST]
+Accept Role Invitation by role invite hash
+
++ Request
+
+    + Headers
+
```
Comment (MeghalBisht) [inline docs/api/blueprint/role/role_invites.apib:272]:
```suggestion
           
           Accept: application/vnd.api+json
           
```

File: docs/api/blueprint/role/role_invites.apib [hunk 2]
```diff
@@ -233,3 +233,61 @@ Delete a single role invite.
...
+## User Email Details By Role Invite [/v1/role_invites/user]
+
+### Get User Email Detail By Role Invite Token [POST]
+Get a single user email by sending role invite hash.
+
++ Request
+
+    + Headers
+
+            Content-Type: application/json
```
Comment (ranjitkshah) [inline docs/api/blueprint/role/role_invites.apib:246]:
```suggestion
            Content-Type: application/vnd.api+json
```

File: docs/api/blueprint/role/role_invites.apib [hunk 3]
```diff
@@ -233,3 +233,61 @@ Delete a single role invite.
...
+## Accept Role Invite [/v1/role_invites/accept-invite]
+
+### Accept Role Invite using Token [POST]
+Accept Role Invitation by role invite hash
+
++ Request
+
+    + Headers
+           
+            Accept: application/vnd.api+json
```
Comment (ranjitkshah) [inline docs/api/blueprint/role/role_invites.apib:273]:
```suggestion
            Content-Type: application/vnd.api+json
```

**Review by ranjitkshah — CHANGES_REQUESTED** [review]

**Review by ranjitkshah — DISMISSED** [review]

**Review by progmatic-99 — CHANGES_REQUESTED** [review]
Dredd tests are failing, need to check what's wrong

**Review by pc-beast — APPROVED** [review]

**Review by pc-beast — APPROVED** [review]

**Pull request merged**